<launch>

    <!-- Arguments -->
    <!-- default="/camera/image_raw" -->
    <arg name="video_feed_topic" value="/rms/camera1/image_raw"/>
    <arg name="focal_length_pxl" value="580"/>

    <arg name="desired_point_1_x" value="360"/>
    <arg name="desired_point_1_y" value="170"/>
    <arg name="desired_point_2_x" value="360"/>
    <arg name="desired_point_2_y" value="290"/>
    <arg name="desired_point_3_x" value="360"/>
    <arg name="desired_point_3_y" value="410"/>

    <!-- Node -->
    <node pkg="visual_servoing" type="visual_servoing_script.py" name="visual_servoing_node" output="screen">
        <param name="~video_feed_topic" type="str" value="$(arg video_feed_topic)"/>
        <param name="~focal_length_pxl" type="int" value="$(arg focal_length_pxl)"/>
        
        <param name="~desired_point_1_x" type="int" value="$(arg desired_point_1_x)"/>
        <param name="~desired_point_1_y" type="int" value="$(arg desired_point_1_y)"/>
        <param name="~desired_point_2_x" type="int" value="$(arg desired_point_2_x)"/>
        <param name="~desired_point_2_y" type="int" value="$(arg desired_point_2_y)"/>
        <param name="~desired_point_3_x" type="int" value="$(arg desired_point_3_x)"/>
        <param name="~desired_point_3_y" type="int" value="$(arg desired_point_3_y)"/>
    </node>

    <node name="image_view_vspub" pkg="image_view" type="image_view" respawn="false" output="screen">
        <remap from="image" to="/vs/robot_0/image_pub"/>
    </node>

    <!-- 
        <node name="image_view_feed" pkg="image_view" type="image_view" respawn="false" output="screen">
            <remap from="image" to="$(arg camera_feed_topic)"/>
        </node>
     -->
    <!--
        <group ns="/vs">
            <arg name="video_feed_topic" value="/rms/camera1/image_raw"/>
        </group>
    -->
</launch>
